include/alpaka/rand/RandGenericSycl.hpp

```diff
@@ -0,0 +1,248 @@
+            }
+            result_type operator()()
+            {
+                oneapi::dpl::uniform_real_distribution<float> distr;
+                return distr(rng_engine);
+            }
+        };
+    } // namespace engine::sycl_rand
+
+    namespace distribution::sycl_rand
+    {
+        //! The SYCL random number float normal distribution.
+        template<>
+        class NormalReal<float>
+        {
+        public:
+            template<typename TEngine>
+            auto operator()(TEngine& engine) -> float
+            {
+                // Create float uniform_real_distribution distribution
+                oneapi::dpl::normal_distribution<float> distr;
+
+                // Generate float random number
+                return distr(engine.rng_engine);
+            }
+        };
+
+        //! The SYCL random number double normal distribution.
+        template<>
+        class NormalReal<double>
+        {
+        public:
+            template<typename TEngine>
+            auto operator()(TEngine& engine) -> double
+            {
+                // Create float uniform_real_distribution distribution
+                oneapi::dpl::normal_distribution<double> distr;
+
+                // Generate float random number
+                return distr(engine.rng_engine);
+            }
+        };
+
+        //! The SYCL random number float uniform distribution.
+        template<>
+        class UniformReal<float>
+        {
+        public:
+            template<typename TEngine>
+            auto operator()(TEngine& engine) -> float
+            {
+                // Create float uniform_real_distribution distribution
+                oneapi::dpl::uniform_real_distribution<float> distr;
+
+                // Generate float random number
+                return distr(engine.rng_engine);
+            }
+        };
+
+        //! The SYCL random number float uniform distribution.
+        template<>
+        class UniformReal<double>
+        {
+        public:
+            template<typename TEngine>
+            auto operator()(TEngine& engine) -> double
+            {
+                // Create float uniform_real_distribution distribution
+                oneapi::dpl::uniform_real_distribution<double> distr;
+
+                // Generate float random number
+                return distr(engine.rng_engine);
+            }
+        };
+
+        //! The SYCL random number unsigned integer uniform distribution.
+        template<>
+        class UniformUint<unsigned int>
+        {
+        public:
+            template<typename TEngine>
+            auto operator()(TEngine& engine) -> unsigned int
+            {
+                // Create float uniform_real_distribution distribution
+                oneapi::dpl::uniform_int_distribution<unsigned int> distr;
+
+                // Generate float random number
+                return distr(engine.rng_engine);
+            }
+        };
```

Comment:
If you just name the above classes and this one here `template <typename T> Uniform`, then you can unify all 3 implementations using an `if constexpr(std::is_floating_point_v<T>)` inside to choose the distribution.